jobs/tests/test_edit_job.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+from jobs.forms import JobForm
+# from jobs.views import UpdateJobsView
+# from jobs.models import Job
+
+
+@pytest.mark.django_db
+def test_view_edit_job_form(client, user_HR, job_1):
+    client.force_login(user_HR)
+    response = client.get(f'/jobs/edit/{job_1.id}')
+    assert response.status_code == 200
+
+    form = response.context["form"]
+    assert isinstance(form, JobForm)
+
+
+@pytest.mark.django_db
+def test_load_edit_form(client, user_HR, job_1):
+    client.force_login(user_HR)
+    response = client.get(f'/jobs/edit/{job_1.id}')
+    assert response.status_code == 200
+
+    form_load_data = response.context['form'].initial
+    assert form_load_data['company_name'] == job_1.company_name
+    assert form_load_data['title'] == job_1.title
+    assert form_load_data['location'] == job_1.location
+    assert form_load_data['description'] == job_1.description
+    assert str(form_load_data['post_until']) == job_1.post_until
+    assert form_load_data['is_active'] == job_1.is_active
+    assert form_load_data['apply_link'] == job_1.apply_link
```

Comment:
Having the test assert on each property is hard to maintain, any change to the properties will require changing the test as well which kind of missing the point in having the tests in the first place.
In other words, the test doesn't need to be aware of the implementation details 
in a way that if the implementation changes the test should still cover it.

Reply:
@ormergi Do you suggest we user the meta data to retrieve the list of properties so we can assert without knowing the properties? 
Will this method help(Add to jobs/models.py)? it returns a list of all of the instance's property names
@property
    def fields(self):
        return [ f.name for f in self._meta.fields]
        
to use it in the test you can type: fieldslist = job_1.fields (returns a list of the fields of the model)